### PR TITLE
Issue #141: Support CKA_ALWAYS_AUTHENTICATE for private key objects.

### PR DESCRIPTION
**Pull request by bellgrim**

### CODE REVIEW

File: src/lib/test/ObjectTests.cpp [hunk 1]
```diff
@@ -1975,12 +2103,12 @@ void ObjectTests::testAllowedMechanisms()
 
 	CK_KEY_TYPE keyType = CKK_GENERIC_SECRET;
 	CK_OBJECT_CLASS secretClass = CKO_SECRET_KEY;
-	CK_BYTE key[64];
+	CK_BYTE key[65] = { "0000000000000000000000000000000000000000000000000000000000000000" };
```
Comment (Jakuje) [inline src/lib/test/ObjectTests.cpp:2106]: This looks like some separate change unrelated to the actual issue. Can you put it into separate commit?

**Comment by jariq** [comment]
I did not review the code but I have tested following scenarios:

- Signature creation with `CKM_RSA_PKCS` and `CKM_RSA_PKCS_PSS` mechanisms
  - using RSA key that has `CKA_ALWAYS_AUTENTICATE` attribute set to `CK_FALSE`
  - using RSA key that has `CKA_ALWAYS_AUTENTICATE` attribute set to `CK_TRUE`  
- Signature creation with `CKM_ECDSA` mechanism
  - using EC key that has `CKA_ALWAYS_AUTENTICATE` attribute set to `CK_FALSE`
  - using EC key that has `CKA_ALWAYS_AUTENTICATE` attribute set to `CK_TRUE`

Everything worked as expected. Good job 👍

File: src/lib/test/ObjectTests.cpp (hunk 1 above)
Comment (bellgrim) [reply src/lib/test/ObjectTests.cpp:2106]: This was to fix a Valgrind memory error so that there were no issues when running the ObjectTests. Have split this into two commits.
